### PR TITLE
todo-mvc: Deal with app refactor

### PR DESCRIPTION
**Pull request by novemberborn**
Compatibility with changes in <https://github.com/dojo/app/pull/62>.

### CODE REVIEW

File: todo-mvc/src/main.ts [hunk 1]
```diff
@@ -155,4 +160,9 @@ app.loadDefinition({
 	]
 });
 
-app.realize(document.body);
+// Try to use the native promise so the browser can report unhandled rejections.
+const P = (<any> window).Promise || ShimPromise;
```
Comment (agubler) [inline todo-mvc/src/main.ts]: Can you use `core`'s `global` class here?

Comment (agubler) [inline todo-mvc/src/main.ts]: Also the naming of `P` can probably be improved

File: todo-mvc/src/main.ts [hunk 2]
```diff
@@ -89,12 +88,18 @@ app.registerStore('todo-store', todoStore);
...
 Object.keys(storeTodoActions).forEach((actionName) => {
 	const action: AnyAction = (<any> storeTodoActions)[actionName];
-	app.registerAction(actionName, action);
+	action.configure(app.registryProvider);
 });
 
 Object.keys(uiTodoActions).forEach((actionName) => {
 	const action: AnyAction = (<any> uiTodoActions)[actionName];
-	app.registerAction(actionName, action);
+	if (actionName === 'filter') {
```
Comment (agubler) [inline todo-mvc/src/main.ts]: Do you think we should extract these to a separate action class? This logic seems like it could be brittle.

Comment (novemberborn) [inline todo-mvc/src/main.ts]: I've been improving it a lot in my declarative branch, this workaround should be OK for now.

File: todo-mvc/src/main.ts (hunk 1 above)
Comment (novemberborn) [inline todo-mvc/src/main.ts]: Is it better with the fixup?

**Comment by novemberborn** [comment]
Closing in favor of #56.

**Pull request closed without merging**